tests/ndarrays/test_ndarrays.c

```diff
@@ -1,459 +1,459 @@
+#include <unistd.h>
+
+
+#define getname(X) #X
+#define my_assert(X , Y, dscr) _Generic((X), double: assert_double,\
+                            float: assert_float,\
+                            int32_t: assert_int32,\
+                            double complex : assert_cdouble,\
+                            default: assert_ns)(X , Y, getname(X), getname(Y), dscr, __func__, __FILE__, __LINE__)
+
+void assert_double(double v1 , double v2,
+        const char *v1_name, const char *v2_name,const char *dscr,
+        const char * func, const char *file, int32_t line)
+{
+    if (v1 != v2)
+    {
+        printf("[FAIL] %s:%d:%s\n", file, line, func);
+        printf("[INFO] %s:%f != %s:%f\n", v1_name, v1, v2_name,v2);
+        printf("[DSCR] %s\n", dscr);
+        return ;
+    }
+    printf("[PASS] %s:%d:%s\n", file, line, func);
+    printf("[DSCR] %s\n", dscr);
+}
+
+void assert_float(float v1 , float v2,
+        const char *v1_name, const char *v2_name,const char *dscr,
+        const char * func, const char *file, int32_t line)
+{
+    if (v1 != v2)
+    {
+        printf("[FAIL] %s:%d:%s\n", file, line, func);
+        printf("[INFO] %s:%f != %s:%f\n", v1_name, v1, v2_name,v2);
+        printf("[DSCR] %s\n", dscr);
+        return ;
+    }
+    printf("[PASS] %s:%d:%s\n", file, line, func);
+    printf("[DSCR] %s\n", dscr);
+}
+
+void assert_int32(int32_t v1 , int32_t v2,
+        const char *v1_name, const char *v2_name,const char *dscr,
+        const char * func, const char *file, int32_t line)
+{
+    if (v1 != v2)
+    {
+        printf("[FAIL] %s:%d:%s\n", file, line, func);
+        printf("[INFO] %s:%d != %s:%d\n", v1_name, v1, v2_name,v2);
+        printf("[DSCR] %s\n", dscr);
+        return ;
+    }
+    printf("[PASS] %s:%d:%s\n", file, line, func);
+    printf("[DSCR] %s\n", dscr);
+}
+
+void assert_cdouble(double complex v1 , double complex v2,
+        const char *v1_name, const char *v2_name,const char *dscr,
+        const char * func, const char *file, int32_t line)
+{
+    if (v1 != v2)
+    {
+        printf("[FAIL] %s:%d:%s\n", file, line, func);
+        printf("[INFO] %s:%f+%f*I != %s:%f+%f*I\n", v1_name, creal(v1), cimag(v1), v2_name,creal(v2), cimag(v2));
+        printf("[DSCR] %s\n", dscr);
+        return ;
+    }
+    printf("[PASS] %s:%d:%s\n", file, line, func);
+    printf("[DSCR] %s\n", dscr);
+}
+
+void assert_ns(float v1 , float v2,
+        const char *v1_name, const char *v2_name, const char *dscr,
+        const char * func, const char *file, int32_t line)
+{
+    printf("[FAIL] %s:%d:%s\n", file, line, func);
+    printf("[INFO] not supported type\n");
+    printf("[DSCR] %s\n", dscr);
+}
+
+int32_t test_indexing_int32(void)
+{
+    int32_t m_1[] = {2, 3, 5, 5, 6, 7, 10, 11,
+                12, 260, 6, 8, 8, 0, 45, 0,
+                1, 0, 0, 1, 200, 33, 5, 57,
+                62, 70, 103, 141, 122, 26, 36, 82,
+                8, 10, 4115, 22, 1, 11, 1, 19};
+    int32_t m_1_shape[] = {5, 8};
```

Comment:
why you use a precise type shape?
I think you need t use only int, as mentioned in the header file.

Reply:
You are right, it seems the header still needs some fixing